application/models/Survey.php

```diff
@@ -190,6 +189,18 @@ public function init()
             $this->gsid = null;
             return;
         }
+        if ($this->isNewRecord) {
+            $this->setAttributeDefaults();
```

Comment:
> All of the attributes, in theory, should be picked up from DB. But what if we are missing something? Is this necesarry for the ticket?

Mayve we need to check if attribute->value == null ?

Reply:
Is this necesarry for the ticket? If not, I would just put it on another ticket. Sounds like unnecesarry risk

Reply:
I really don't know … The big part is for user search.

the other default can be set as public var ($htmlemail = 'Y').

But here : i think default are not udated since we add inherit 

We can use rules too to set the default …

Reply:
> I really don't know … The big part is for user search.

What I mean is that these changes (https://github.com/LimeSurvey/LimeSurvey/pull/3785/files#diff-0268c3e4648700cef058f99725fd5364a285bd35701d30cfda885a81ce1c86a4R192-R203) seem not necessary to work on the ticket. Are they?

If not, I wouldn't update that here, but on a DEV branch as to receive more testing and usage before merging the code live.

Reply:
This part are already merged in master

This part is **necessary** to this ticket it's the part of THIS commit : don't set default when not needed 

there are only 

```
        $this->htmlemail = 'Y';
        $this->format = 'G';
        $this->tokenencryptionoptions = '';
```
done without testing (and i think it must be I now)

Else

Uneeded test : https://github.com/LimeSurvey/LimeSurvey/blob/0f38d236fea6ef62fe638a0c12e5cf1913e8d129/application/models/Survey.php#L210
Uneeded validator : https://github.com/LimeSurvey/LimeSurvey/blob/0f38d236fea6ef62fe638a0c12e5cf1913e8d129/application/models/Survey.php#L219
Uneeded test and request https://github.com/LimeSurvey/LimeSurvey/blob/0f38d236fea6ef62fe638a0c12e5cf1913e8d129/application/models/Survey.php#L224

Reply:
> This part are already merged in master

OK, then

Reply:
> OK, then

I think if there are some «old» survey with null as this step : must be fixed by a https://manual.limesurvey.org/Check_data_integrity/en ?
We still need a Survey_check_data_integrity (check integrity for a single survey)